File: C++/bee 1046.cpp
#include <iostream>
#include <iomanip>
#include <cmath>

using namespace std;

int main() {

    int horaInicial, horaFinal;
    cin >> horaInicial >> horaFinal;

    if (horaFinal > horaInicial) {
        cout << "O JOGO DUROU " << horaFinal - horaInicial << " HORA(S)" << endl;
    } else {
        cout << "O JOGO DUROU " << (24 - horaInicial) + horaFinal << " HORA(S)" << endl;
    }


    return 0;
}

File: Rust/bee_1046.rs
use std::io::{self, Read};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();

    let valores: Vec<i32> = input
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();

    let hora_inicial = valores[0];
    let hora_final = valores[1];

    if hora_final > hora_inicial {
        println!("O JOGO DUROU {} HORA(S)", (hora_final - hora_inicial));
    } else {
        println!(
            "O JOGO DUROU {} HORA(S)",
            ((24 - hora_inicial) + hora_final)
        );
    }
}
